contracts/DAITokenWrapper.sol

```diff
@@ -0,0 +1,150 @@
+        
+    }
+
+    /**
+     * @notice  Checking if a non-zero address is provided, otherwise reverts.
+     */
+    modifier notZeroAddress(address tokenAddress) {
+        require(tokenAddress != address(0), "0A"); //zero address
+        _;
+    }
+
+    /**
+     * @notice Conforms to ERC2612. Calls permit on token, which may or may not have a permit function that conforms to ERC2612
+     * @param owner Address of the token owner who is approving tokens to be transferred by spender
+     * @param spender Address of the party who is transferring tokens on owner's behalf
+     * @param value Number of tokens to be transferred
+     * @param deadline Time after which this permission to transfer is no longer valid
+     * @param v Part of the owner's signatue
+     * @param r Part of the owner's signatue
+     * @param s Part of the owner's signatue
+     */
+    function permit(
+        address owner,
+        address spender,
+        uint256 value,
+        uint256 deadline,
+        uint8 v,
+        bytes32 r,
+        bytes32 s
+    ) 
+        external
+        override
+        notZeroAddress(owner)
+        notZeroAddress(spender)
+        whenNotPaused
+    {
+        require(deadline == 0 || block.timestamp <= deadline, "PERMIT_EXPIRED");
+        require(r != bytes32(0) && s != bytes32(0), "INVALID_SIGNATURE_COMPONENTS");
+        uint nonce =  IDAI(daiTokenAddress).nonces(owner);
+        IDAI(daiTokenAddress).permit(owner, spender, nonce, deadline, true, v, r, s);
+        emit LogPermitCalledOnToken(daiTokenAddress, owner, spender, 0);    
+    }
+
+    /**
+     * @notice Set the address of the wrapper contract for the token. The wrapper is used to, for instance, allow the Boson Protocol functions that use permit functionality to work in a uniform way.
+     * @param _tokenAddress Address of the token which will be updated.
+     */
+    function setTokenAddress(address _tokenAddress)
+        external
+        override
+        onlyOwner
+        whenNotPaused
+        notZeroAddress(_tokenAddress)
+    {
+        daiTokenAddress = _tokenAddress;
+        emit LogTokenAddressChanged(_tokenAddress, owner());
+    }
+
+    /**
+     * @notice Get the address of the token wrapped by this contract
+     * @return Address of the token wrapper contract
+     */
+    function getTokenAddress()
+        external
+        view
+        override
+        returns (address)
+    {
+        return daiTokenAddress;
+    }
+
+    /**
+     * @notice Pause the Cashier && the Voucher Kernel contracts in case of emergency.
+     * All functions related to creating new batch, requestVoucher or withdraw will be paused, hence cannot be executed.
+     * There is special function for withdrawing funds if contract is paused.
+     */
+    function pause() 
+        external 
+        override
+        onlyOwner
+    {
+        
+        _pause();
+    }
+
+    /**
+     * @notice Unpause the Cashier && the Voucher Kernel contracts.
```

Comment:
Is the comment correct?
I'm surprised seeing mentions of the Cashier and VoucherKernel contracts
(same in pause() method comments, above)

Reply:
This will be removed when the `unpause` function is removed

Reply:
Pauable removed